File: 백준/Silver/10828. 스택/스택.cc
#include <stdio.h>
#include <iostream>
#include <string>
#include <stack>
int main(){
    std::cin.tie(0);
    std::ios::sync_with_stdio(0);
    std::stack <int> stack ={};
    int n{},num{};
    std::cin >> n ;
    std::string str{};
    for(int i=0; i<n; i++){
        std::cin >> str;
        if(str=="push"){
            std::cin >> num;
            stack.push(num);
        }
        if(str=="pop"){
            if(!stack.empty()){
                std::cout << stack.top()<< "\n";
                stack.pop();
            }
            else{
                std::cout << -1<< "\n";
            }
        }
        if(str=="size"){
            std::cout << stack.size()<< "\n";
        }
        if(str=="empty"){
            if(stack.empty()){
                std::cout << 1<< "\n";
            }
            else{
                std::cout << 0<< "\n";
            }
        }
        if(str=="top"){
            if(stack.empty()){
                std::cout << -1 <<"\n";
            }
            else{
                std::cout << stack.top()<< "\n";
            }
        }
    }
}

File: 백준/Silver/10828. 스택/스택.py
import sys
stack=[]
n=int(input())
for i in range(n):
    line=sys.stdin.readline().split()
    if(line[0]=="push"):
        stack.append(line[1])
    elif(line[0]=="pop"):
        if(stack):
            print(stack.pop())
        else:
            print(-1)
    elif(line[0]=="size"):
        print(len(stack))
    elif(line[0]=="empty"):
        if(stack):
            print(0)
        else:
            print(1)
    elif(line[0]=="top"):
        if(stack):
            print(stack[len(stack)-1])
        else:
            print(-1)